Tasks/ServiceFabricUpdateManifests/Update-DockerImageSettings.psm1

```diff
@@ -0,0 +1,61 @@
+function Update-DockerImageSettings
+{
+    [CmdletBinding()]
+    Param()
+
+    Trace-VstsEnteringInvocation $MyInvocation
+    try
+    {
+        $imageDigestsFilePathSearchPattern = Get-VstsInput -Name imageDigestsPath -Require
+        $imageDigestsPath = Find-VstsFiles -LegacyPattern $imageDigestsFilePathSearchPattern
+
+        $appPackagePathSearchPattern = (Get-VstsInput -Name applicationPackagePath -Require)
+        $appPackagePath = Find-VstsFiles -LegacyPattern $appPackagePathSearchPattern -IncludeDirectories
+
+        # Collect the image digest values for the images
+        $imageDigestValues = (Get-Content -Path $imageDigestsPath).Replace("`r`n", "`n").Split("`n")
+        $imageNameToDigestMapping = @{}
+        foreach ($imageDigestValue in $imageDigestValues)
+        {
+            $slashIndex = $imageDigestValue.IndexOf("/")
+            $hashSeparatorIndex = $imageDigestValue.IndexOf("@")
+            if ($slashIndex -lt 0 -or $hashSeparatorIndex -lt 0)
+            {
+                throw (Get-VstsLocString -Key InvalidImageDigestValue -ArgumentList @($imageDigestValue, $imageDigestsPath))
+            }
+            $imageName = $imageDigestValue.Substring($slashIndex + 1, $hashSeparatorIndex - $slashIndex - 1)
+            $imageNameToDigestMapping[$imageName] = $imageDigestValue
+        }
+
+        # Update the service manifests to use the digest-qualified image names
+        $appManifestPath = Join-Path $appPackagePath "ApplicationManifest.xml"
+        $appManifestXml = [xml](Get-Content $appManifestPath)
+        foreach ($serviceManifestImport in $appManifestXml.ApplicationManifest.ServiceManifestImport)
+        {
+            $serviceManifestPath = [System.IO.Path]::Combine($appPackagePath, $serviceManifestImport.ServiceManifestRef.ServiceManifestName, "ServiceManifest.xml")
+            $serviceManifestXml = [xml](Get-Content $serviceManifestPath)
+
+            $hasUpdates = $false
+            $codePackages = @($serviceManifestXml.ServiceManifest.CodePackage)
+            foreach ($codePackage in $codePackages)
+            {
+                if ($codePackage.EntryPoint -and $codePackage.EntryPoint.ContainerHost -and $codePackage.EntryPoint.ContainerHost.ImageName)
+                {
+                    $digest = $imageNameToDigestMapping[$codePackage.EntryPoint.ContainerHost.ImageName]
+                    if ($digest)
+                    {
+                        $codePackage.EntryPoint.ContainerHost.ImageName = $digest
```

Comment:
I agree to this in principal. However, updating image and updating version for that image service(in same manifest) are very thinly separated operations. And for a user, it will be difficult to visualize that for updating image related information in manifest, he needs to use same task twice.
However IMO my suggestion is only important if makes sense from implementation point of view to do so.
This becomes much less serious issue if user is also updating other services' manifests. The it is easy to visualize that task needs to be used twice.